toolchain/check/import_cpp.cpp

```diff
@@ -141,64 +138,75 @@ auto ImportCppFiles(Context& context, llvm::StringRef importing_file_path,
   name_scope.set_is_closed_import(true);
   name_scope.set_is_cpp_scope(true);
 
-  *ast = std::move(generated_ast);
-  context.sem_ir().set_cpp_ast(ast->get());
+  context.sem_ir().set_cpp_ast(generated_ast.get());
 
   if (ast_has_error) {
     name_scope.set_has_error();
   }
+
+  return std::move(generated_ast);
 }
 
-static auto ClangLookup(Context& context, SemIR::NameId name_id)
+// Lookups the given name in the Clang AST. Returns the lookup result if lookup
+// was successful.
+static auto ClangLookup(Context& context, SemIR::LocId loc_id,
+                        SemIR::NameId name_id)
     -> std::optional<clang::LookupResult> {
   clang::ASTUnit* ast = context.sem_ir().cpp_ast();
   CARBON_CHECK(ast);
-  CARBON_CHECK(ast->hasSema());
   clang::Sema& sema = ast->getSema();
 
-  auto name = context.names().GetAsStringIfIdentifier(name_id);
-  CARBON_CHECK(name);
+  llvm::StringRef name = context.names().GetIRBaseName(name_id);
 
   clang::LookupResult lookup(
       sema,
       clang::DeclarationNameInfo(
           clang::DeclarationName(
-              sema.getPreprocessor().getIdentifierInfo(*name)),
+              sema.getPreprocessor().getIdentifierInfo(name)),
           clang::SourceLocation()),
       clang::Sema::LookupNameKind::LookupOrdinaryName);
 
-  if (!sema.LookupQualifiedName(
-          lookup, ast->getASTContext().getTranslationUnitDecl())) {
+  bool found = sema.LookupQualifiedName(
+      lookup, ast->getASTContext().getTranslationUnitDecl());
+
+  if (lookup.isClassLookup()) {
+    // TODO: When support class lookup, also check access.
+    context.TODO(loc_id, "Unsupported: Lookup in Class");
```

Comment:
Maybe (given the "unsupported" string):

```suggestion
    context.TODO(loc_id, "Unsupported: Lookup in Class");
    return std::nullopt;
```

Reply:
Done.